### PR TITLE
[core] Add resource idle time to resource report from node. 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

This adds the information of `resources_idle_duration_ms` into periodical node's resource reporting to GCS. 

This PR tracks the idle time of each resource id:
- When a resource is not used (available == total): it records the timestamp.
- When a resource is being used (available < total): it clears the timestamp. 

So a node's resource idle time is the maximal timestamp of all the resources' idle time. And it's idle duration is the duration between current time and the idle time. 

TODO: I will do the PR where use this time for node states (dead time, idle time, and running time) in a follow-up. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/raylet/scheduling/local_resource_manager.cc [hunk 1]
```diff
@@ -114,6 +124,8 @@ bool LocalResourceManager::AllocateResourceInstances(
   allocation->resize(available.size());
   FixedPoint remaining_demand = demand;
 
+  RAY_LOG(INFO) << "Allocating " << demand << " from " << available.size();
```
Comment (scv119) [inline src/ray/raylet/scheduling/local_resource_manager.cc:127]: debug?

File: src/ray/raylet/scheduling/local_resource_manager.cc [hunk 2]
```diff
@@ -256,6 +282,34 @@ std::vector<double> LocalResourceManager::SubtractResourceInstances(
...
+
+absl::optional<absl::Time> LocalResourceManager::GetResourceIdleTime() const {
+  // If all the resources are idle.
+  absl::Time all_idle_time = absl::InfinitePast();
+
+  for (const auto &iter : resources_last_idle_time_) {
+    const auto &idle_time_or_busy = iter.second;
+
+    if (idle_time_or_busy == absl::nullopt) {
+      // One resource is busy, entire resource
```
Comment (scv119) [inline src/ray/raylet/scheduling/local_resource_manager.cc:303]: comment looks broken

File: src/ray/raylet/scheduling/local_resource_manager.cc [hunk 3]
```diff
@@ -403,6 +470,10 @@ std::optional<syncer::RaySyncMessage> LocalResourceManager::CreateSyncMessage(
 
   resources_data.set_resources_available_changed(true);
 
+  const auto now = absl::Now();
+  resources_data.set_idle_duration_ms(
```
Comment (scv119) [inline src/ray/raylet/scheduling/local_resource_manager.cc:474]: hmm should we use -1 or something else instead? using to indicate resource not idle could be confusing.

Comment (rickyyx) [reply src/ray/raylet/scheduling/local_resource_manager.cc:474]: So it's currently using `0` to indicate not idle. If we use `-1` for not idle, technically what will be `0` ?

**Review by scv119 — APPROVED** [review]

**Pull request merged**